nxt_editor/dockwidgets/property_editor.py

```diff
@@ -1054,12 +1089,13 @@ def exec_context_menu(self):
                                                                 INTERNAL_ATTRS.EXECUTE_IN,
                                                                 self.stage_model.comp_layer)
         tgt_path = self.stage_model.target_layer.real_path
-        if src_path == self.node_path and layer == tgt_path:
-            self.localize_exec_path_action.setEnabled(False)
-            self.revert_exec_path_action.setEnabled(True)
-        else:
-            self.localize_exec_path_action.setEnabled(True)
-            self.revert_exec_path_action.setEnabled(False)
+        if not self.locked:
+            if src_path == self.node_path and layer == tgt_path:
+                self.localize_exec_path_action.setEnabled(False)
+                self.revert_exec_path_action.setEnabled(True)
+            else:
+                self.localize_exec_path_action.setEnabled(True)
+                self.revert_exec_path_action.setEnabled(False)
```

Comment:
I'm concerned to see that there's a case where we wouldn't explicitly enable or disable these two actions in this menu. If it's locked we would want them both disabled right?

This removes the branching and integrates lock into making sure these are always enabled or disabled intentionally. 
```suggestion
        exec_is_path_local = (src_path == self.node_path) and (layer == tgt_path)
        self.localize_exec_path_action.setEnabled(not exec_is_path_local and not self.locked)
        self.revert_exec_path_action.setEnabled(exec_is_path_local and not self.locked)
```

Reply:
Very nice I like.